internal/http/services/owncloud/ocs/conversions/main.go

```diff
@@ -273,15 +275,38 @@ func AsCS3Permissions(p int, rp *provider.ResourcePermissions) *provider.Resourc
 	return rp
 }
 
+// UserServiceUserIDResolver returns a userId resolver using the UserService
+func UserServiceUserIDResolver(ctx context.Context, client gatewayv1beta1.GatewayAPIClient) func(*userpb.UserId) (string, error) {
+	return func(id *userpb.UserId) (string, error) {
+		resp, err := client.GetUser(ctx, &userpb.GetUserRequest{UserId: id})
```

Comment:
this must be cached, otherwise we will make an rpc to the gateway and then to the user provider, probably to ldap ... and back, several times per file list entry ...

Reply:
Ugh... I think this is also causing the tests to fail... The ldap server gets overwhelmed and then crashes...
Locally the pipelines work... :

Reply:
I implemented the username lookup similar to the display name lookup which you implemented. Including the caching.